src/sentry/integrations/api/serializers/models/data_forwarder.py

```diff
@@ -0,0 +1,87 @@
+from collections import defaultdict
+from collections.abc import Mapping, Sequence
+from typing import Any
+
+from django.contrib.auth.models import AnonymousUser
+
+from sentry.api.serializers import Serializer, register, serialize
+from sentry.integrations.models.data_forwarder import DataForwarder
+from sentry.integrations.models.data_forwarder_project import DataForwarderProject
+from sentry.users.models.user import User
+from sentry.users.services.user import RpcUser
+
+
+@register(DataForwarder)
+class DataForwarderSerializer(Serializer):
+    def get_attrs(
+        self,
+        item_list: Sequence[DataForwarder],
+        user: User | RpcUser | AnonymousUser,
+        **kwargs: Any,
+    ) -> dict[DataForwarder, dict[str, Any]]:
+        project_configs = DataForwarderProject.objects.filter(
+            data_forwarder__in=item_list
+        ).select_related("project", "data_forwarder")
+        project_config_attrs = defaultdict(set)
+        for project_config in project_configs:
+            project_config_attrs[project_config.data_forwarder].add(project_config)
+
+        return {
+            data_forwarder: {"project_configs": project_config_attrs.get(data_forwarder, set())}
+            for data_forwarder in item_list
+        }
+
+    def serialize(
+        self,
+        obj: DataForwarder,
+        attrs: Mapping[str, Any],
+        user: User | RpcUser | AnonymousUser,
+        **kwargs: Any,
+    ) -> dict[str, Any]:
+        project_configs = attrs.get("project_configs", set())
+
+        return {
+            "id": str(obj.id),
+            "organizationId": str(obj.organization_id),
+            "isEnabled": obj.is_enabled,
+            "enrollNewProjects": obj.enroll_new_projects,
+            "enrolledProjects": [
+                {
+                    "id": str(project_config.project_id),
+                    "slug": project_config.project.slug,
+                    "platform": project_config.project.platform,
+                }
+                for project_config in project_configs
+            ],
+            "provider": obj.provider,
+            "config": obj.config,
+            "projectConfigs": [
+                serialize(project_config, user=user) for project_config in project_configs
+            ],
+            "dateAdded": obj.date_added,
+            "dateUpdated": obj.date_updated,
+        }
+
+
+@register(DataForwarderProject)
+class DataForwarderProjectSerializer(Serializer):
+    def serialize(
+        self,
+        obj: DataForwarderProject,
+        attrs: Mapping[str, Any],
+        user: User | RpcUser | AnonymousUser,
+        **kwargs: Any,
+    ) -> dict[str, Any]:
+        return {
+            "id": str(obj.id),
+            "dataForwarderId": str(obj.data_forwarder_id),
+            "project": {
+                "id": str(obj.project_id),
+                "slug": obj.project.slug,
+                "platform": obj.project.platform,
+            },
```

Comment:
These wont necessarily have the same prefetch guarentees though, I think you'll have to do some `get_attrs` magic again here like you did above to avoid N+1 queries.

Reply:
Actually i guess if its always nested it will always be prefetched, it only N+1s if you call it separately 🤔  i think you can leave this as is in that case